quickwit-metastore/src/tests.rs

```diff
@@ -113,13 +92,17 @@ pub mod test_suite {
             .create_index(index_metadata.clone())
             .await
             .unwrap();
+        assert_eq!(metastore.list_all_splits(index_id).await.unwrap().len(), 0);
```

Comment:
Do we need an `index_exists` in the public API?

Reply:
Maybe in the future but right now there is nowhere in the use of Metastore where we need to check if e this implicitly will tell us if index does not exist.